src/app/GitCommands/ServiceContainerRegistry.cs

```diff
@@ -8,6 +8,6 @@ public static class ServiceContainerRegistry
 {
     public static void RegisterServices(ServiceContainer serviceContainer)
     {
-        serviceContainer.AddService<ISubmoduleStatusProvider>(new SubmoduleStatusProvider());
+        serviceContainer.AddService<ISubmoduleStatusProvider>(new SubmoduleStatusProvider((path) => new GitModule(path)));
```

Comment:
[nit] I think the parens are unnecessary -> `...new SubmoduleStatusProvider(path => new GitModule(path))`